Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/AlternativeIdService.java

```diff
@@ -0,0 +1,90 @@
+package edu.asu.conceptpower.app.core.impl;
+
+import java.util.Collection;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.conceptpower.app.core.IAlternativeIdService;
+import edu.asu.conceptpower.app.core.IConceptTypesService;
+import edu.asu.conceptpower.app.core.IConceptTypesService.TypeOfId;
+import edu.asu.conceptpower.core.ConceptEntry;
+
+/**
+ * This helper class is used for adding alternative ids to the concept entry.
+ * 
+ * @author karthikeyanmohan
+ *
+ */
+@Service
+public class AlternativeIdService implements IAlternativeIdService {
+
+    @Autowired
+    private IConceptTypesService conceptTypesService;
+
+    /**
+     * This method adds all alternative ids to the given concept entry including
+     * the id that was used to search for a concept (passed as method
+     * parameter).
+     * 
+     * If id or concept entry passed to the method is null, then method just
+     * returns and no changes are made.
+     * 
+     * The queriedId parameter will be added to the alternative id list if the
+     * queriedId is of Generic wordnet concept type.
```

Comment:
paragraph 1 and 3 are generally true (should be true for any implementation of the interface), so they should be on the interface. However, paragraph 1 and 3 are also contradicting:
Para 1: "adds all alternative ids to the given concept entry including the id that was..."
Para 3: "will be added to the alternative id list if the queriedId is of Generic wordnet concept type."
Which one is it?